app/assets/javascripts/test/enzyme/e2e-setup.js

```diff
@@ -0,0 +1,88 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+// @flow
+import fs from "fs";
+import himalaya from "himalaya";
+import fetch, { Headers, Request, Response, FetchError } from "node-fetch";
+import jsRoutes from "./jsRoutes";
+
+const requests = [];
+const minimumWait = 10;
+function waitForAllRequests() {
+  return new Promise(resolve => {
+    let length = requests.length;
```

Comment:
Should we defensively copy the `requests` array?

Reply:
Shouldn't saving the length be enough?

Reply:
true, if you never remove something from it